internal/handlers/order.go

```diff
@@ -194,21 +220,82 @@ func (o Order) ListOrderFunc(repository domain.OrderRepository) orders.GetAllOrd
 		if p.Status != nil {
 			_, ok := domain.AllOrderStatuses[*p.Status]
 			if !ok {
-				return orders.NewGetAllOrdersDefault(http.StatusBadRequest).
+				return orders.NewGetUserOrdersDefault(http.StatusBadRequest).
 					WithPayload(buildStringPayload(fmt.Sprintf("Invalid order status '%v'", *p.Status)))
 			}
 		}
 
 		total, err := repository.OrdersTotal(ctx, userID)
 		if err != nil {
 			o.logger.Error("Error while getting total of all user's orders", zap.Error(err))
+			return orders.NewGetUserOrdersDefault(http.StatusInternalServerError).
+				WithPayload(buildErrorPayload(err))
+		}
+
+		var items []*ent.Order
+		if total > 0 {
+			items, err = repository.List(ctx, &userID, orderFilter)
+			if err != nil {
+				o.logger.Error("list items failed", zap.Error(err))
+				return orders.NewGetUserOrdersDefault(http.StatusInternalServerError).WithPayload(buildErrorPayload(err))
+			}
+		}
+
+		mappedOrders, err := mapUserOrdersToResponse(items, o.logger)
+		if err != nil {
+			o.logger.Error("map orders to response failed", zap.Error(err))
+			return orders.NewGetUserOrdersDefault(http.StatusInternalServerError).WithPayload(buildErrorPayload(err))
+		}
+		totalOrders := int64(total)
+		listOrders := &models.UserOrdersList{
+			Items: mappedOrders,
+			Total: &totalOrders,
+		}
+		return orders.NewGetUserOrdersOK().WithPayload(listOrders)
+	}
+}
+
+func (o Order) ListAllOrdersFunc(repository domain.OrderRepository) orders.GetAllOrdersHandlerFunc {
+	return func(p orders.GetAllOrdersParams, _ *models.Principal) middleware.Responder {
+		ctx := p.HTTPRequest.Context()
+		limit := utils.GetValueByPointerOrDefaultValue(p.Limit, math.MaxInt)
+		offset := utils.GetValueByPointerOrDefaultValue(p.Offset, 0)
+		orderBy := utils.GetValueByPointerOrDefaultValue(p.OrderBy, utils.AscOrder)
+		orderColumn := utils.GetValueByPointerOrDefaultValue(p.OrderColumn, order.FieldID)
+
+		orderFilter := domain.OrderFilter{
+			Filter: domain.Filter{
+				Limit:       int(limit),
+				Offset:      int(offset),
+				OrderBy:     orderBy,
+				OrderColumn: orderColumn,
+			},
+		}
```

Comment:
I found that this part repeats in 2 functions ListAllOrdersFunc and ListAllOrdersFunc. I think you should use DRY principle here.

Reply:
I didn't find any solution here because of different type of `p` in the mentioned handlers. Please share your ideas if any